charmpath.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmrepo
+
+import (
+	"os"
+	"path/filepath"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+)
+
+// CharmPath is used to access a charm at a given path.
+type CharmPath interface {
+	// Charm returns the charm represented by this path.
+	// If the series is not specified, the charm's default
+	// series is used, if any. Otherwise, the series is
+	// validated against those the charm declares it supports.
+	Charm(series string) (charm.Charm, *charm.URL, error)
+}
+
+// charmPath represents a path (eg local directory) containing a
+// single charm.
+type charmPath struct {
+	Name      string
+	CharmInfo charm.Charm
+}
+
+var _ CharmPath = (*charmPath)(nil)
+
+// NewCharmPath creates and returns an instance used to
+// open a charm located at the given path.
+func NewCharmPath(path string) (CharmPath, error) {
```

Comment:
Same here - I can't see what the CharmPath interface adds over just returning (charm.Charm, *charm.URL, error) here.